src/pdf.sandbox.external.js

```diff
@@ -0,0 +1,174 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// In mozilla-central, this file is loaded as non-module script,
+// so it mustn't have any dependencies.
+
+class SandboxSupportBase {
+  /**
+   * @param {DOMWindow} win
+   */
+  constructor(win, sandbox) {
```

Comment:
The `sandbox` parameter is not documented. By trying to document you may discover that it's not really straightforward to document it because the `sandbox` object is different here in quickjs and over there in m-c.

You're currently calling `this.sandbox.callFunction` directly, but the problem with this is that it forces users of `SandboxSupportBase` to declare a `callFunction` method on an undocumented parameter. This `SandboxSupportBase` class should not need to know anything about the sandbox implementation, as that could indicate a violation of the separation of concerns.

In my initial suggestion at https://github.com/mozilla/pdf.js/pull/12689#discussion_r542377524, I intentionally excluded `sandbox` from the signature of the `SandboxSupportBase`, so that users/subclasses can change the implementation without depending on `SandboxSupportBase` (and vice versa).

I suggest to do the same, by replacing `self.sandbox.callFunction` with `self.callSandboxFunction`, and letting subclasses defining their own implementation of `callSandboxFunction`, which could look like this:

```
constructor(win) {
  super(win);
  this.commFunc = null; // Will be assigned after the sandbox is initialized
}
callSandboxFunction(name, args) {
  args = this.support.exportValueToSandbox(args);
  try {
    this.commFunc(name, args);
  } catch (e) {
    Cu.reportError(e); // in m-c, or console.error(e) in quickjs
  }
}
```

note: I removed the typeof `name` check and the `commFunc` null check, which is present in your current implementation, because 1) name should never not be a string and 2) the function should never be called when `commFunc` is not set. Adding checks for these and silently ignoring them results in less robust code, because errors go unnoticed.